### PR TITLE
[Offline base maps] Add offline base map sources to remote and local databases

### PR DESCRIPTION
**Pull request by gino-m**
Work towards #516.

Adding a since repeated field was much more work than expected (286 lines across 13 files). In large part this is due to how we persist data locally. Filed #517 with a few ideas of how to improve this in the future.

### CODE REVIEW

**Comment by gino-m** [comment]
@scolsen Could you PTAL?

File: gnd/src/main/java/com/google/android/gnd/model/Project.java [hunk 1]
```diff
@@ -39,6 +40,9 @@
   @NonNull
   protected abstract ImmutableMap<String, Layer> getLayerMap();
 
+  @NonNull
+  public abstract ImmutableList<OfflineBaseMapSource> getOfflineBaseMapSources();
```
Comment (scolsen) [inline gnd/src/main/java/com/google/android/gnd/model/Project.java:44]: Cool! What's the envisioned flow for selecting between these? 

Offline Areas -> Add Area -> Select Source -> Download... or something different?

Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/model/Project.java:44]: These will be configured via the web app only - there will likely be an overflow menu with advanced project settings where we can paste the URL.

**Comment by gino-m** [comment]
Thanks for the review!

**Pull request merged**